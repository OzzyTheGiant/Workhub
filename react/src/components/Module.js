import React from "react";
import PropTypes from "prop-types";

const Module = (props) => {
    let toolbarButtons = null; 
    if (props.buttonActions) {
        toolbarButtons = props.buttonActions.map((func, index) => {
            return <button 
                key={index} 
                className={func.name.split(" ")[1] + " toolbar-button button-" + props.displayType} 
                onClick={func}></button>;
        });
    }
	return (
		<div id={props.title + "Module"} className="Module">
			<div id="toolbar">
                <h1>{props.title}</h1>
                <div className="toolbarButtons">{toolbarButtons}</div>
            </div>
			{props.children}
		</div>
	);
};

Module.propTypes = {
    title:PropTypes.string.isRequired,
    displayType:PropTypes.string,
    buttonActions:PropTypes.arrayOf(PropTypes.func)
};

export default Module;