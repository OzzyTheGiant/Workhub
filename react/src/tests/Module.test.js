import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Module from 'components/Module';

describe("Module", () => {
    /* ==== MOCKS ==== */
    const test = jest.fn(function test() {});

    /* ==== COMPONENT ==== */
    let componentWrapper = null;
	const module = <Module title="Sample Module" buttonActions={[test]}><div></div></Module>;

	beforeAll(() => {
		configure({adapter:new Adapter()})
    });
    
    beforeEach(() => componentWrapper = shallow(module));

	it("Should render correctly", () => {	
		expect(toJson(componentWrapper)).toMatchSnapshot();
    });
    
    it("Should execute designated functions when clicking button on toolbar", () => {
        componentWrapper.find("button").at(0).simulate('click');
        expect(test).toBeCalled();
    });
});