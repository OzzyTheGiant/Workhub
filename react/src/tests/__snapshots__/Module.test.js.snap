// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module Should render correctly 1`] = `
<div
  className="Module"
  id="Sample ModuleModule"
>
  <div
    id="toolbar"
  >
    <h1>
      Sample Module
    </h1>
    <div
      className="toolbarButtons"
    >
      <button
        className="undefined toolbar-button button-undefined"
        key="0"
        onClick={[Function]}
      />
    </div>
  </div>
  <div />
</div>
`;
